homeassistant/components/geo_location/demo.py

```diff
@@ -0,0 +1,165 @@
+"""
+Demo platform for the geo location component.
+
+For more details about this platform, please refer to the documentation
+https://home-assistant.io/components/demo/
+"""
+import logging
+import random
+from datetime import timedelta
+from math import pi, cos, sin, radians
+
+from typing import Optional
+
+from homeassistant.components import group
+from homeassistant.components.geo_location import GeoLocationEvent, DOMAIN
+from homeassistant.helpers.entity import generate_entity_id
+from homeassistant.helpers.event import track_time_interval
+from homeassistant import util
+
+_LOGGER = logging.getLogger(__name__)
+
+AVG_KM_PER_DEGREE = 111.0
+DEFAULT_UNIT_OF_MEASUREMENT = "km"
+DEFAULT_UPDATE_INTERVAL = timedelta(minutes=1)
+ENTITY_ID_FORMAT = DOMAIN + '.{}'
+MAX_RADIUS_IN_KM = 50
+NUMBER_OF_DEMO_DEVICES = 5
+
+EVENT_NAMES = ["Bushfire", "Hazard Reduction", "Grass Fire", "Burn off",
+               "Structure Fire", "Fire Alarm", "Thunderstorm", "Tornado",
+               "Cyclone", "Waterspout", "Dust Storm", "Blizzard", "Ice Storm",
+               "Earthquake", "Tsunami"]
+
+
+def setup_platform(hass, config, add_devices, discovery_info=None):
+    """Set up the Demo geo locations."""
+    manager = DemoManager(hass, add_devices)
+    return manager is not None
+
+
+class DemoManager:
+    """Device manager for demo geo location events."""
+
+    def __init__(self, hass, add_devices):
+        """Initialise the demo geo location event manager."""
+        self._hass = hass
+        self._add_devices = add_devices
+        self._name = "Geo Location Demo"
+        self._managed_devices = []
+        self.group = group.Group.create_group(self._hass, self._name,
```

Comment:
We shouldn't group entities in platforms besides the group platform.

Reply:
OK, I'll remove this for now. I still need to get my head around how to implement future features like the zone trigger without any kind of grouping or joint identification of entities that have been generated by a platform.